### ISSUE TITLE
CPU Board attached to the Bebop drone

### ISSUE THREAD

**Issue by mandelyoni** [uzh-rpg/rpg_public_dronet#19]
Hi,

Which CPU Board is recommended to attach to the Bebop drone for running the net? 
If available, please share a link.

Thank you.

**Comment by antonilo** [comment]
Hi,
We are using the Intel UpBoard (https://up-shop.org/27-up) for onboard computation.

**Comment by mandelyoni** [comment]
Thank you for the prompt response.
By the way, are you going to sell the small docking board with Dronet for the crazyflie soon?

**Comment by antonilo** [comment]
We do not plan to sell it since we are still in development phase. While the board can be already [purchased](https://www.pulp-platform.org/), the integrated system will follow in the (hopefully) near future.

**Comment by mandelyoni** [comment]
Hi, thanks for your answer.
Which board, did you mean, can already be purchased? 
Is it straight forward to implement the the Dronet on it?

**Comment by antonilo** [comment]
The PULP board that runs dronet in Ultra-low power can already be purchased. Implementation of the algorithm that is highly non-trivial, but we are currently on it. We will soon share our findings and implementation.

**Comment by mandelyoni** [comment]
Hi Antonilo,

Can you please share an explanation/description of how to connect the Up-board to the Bebop2?

Thanks,  Yoni

**Comment by antonilo** [comment]
I would argue that the easier way to do it is using ROS. This is however not the only one and for sure not the best one in term of performance.

**Comment by mandelyoni** [comment]
Thanks, but I meant how connect the Up-Board physically to the drone (power and communication)?

Thanks again, Yoni

**Comment by antonilo** [comment]
We have an external battery to power the Up-Board and communication happens via WiFi and ROS.
Hope this helps!

**Comment by mandelyoni** [comment]
Hi Antonilo,

Is there a big difference in the performance between running the algorithms on the UpBoard or on a laptop ?
In most of your videos we didn't see the UpBoard connected to your drone? Can you please explain why?

Thank you,  Yoni.

**Comment by antonilo** [comment]
In the original dronet video, computation was done off-board in a laptop. In our last projects, all computation has been shifted to fully on-board: [Deep Drone Racing](https://www.youtube.com/watch?v=8RILnqPxo1s&feature=youtu.be), [Beauty and the Beast](https://www.youtube.com/watch?v=UuQvijZcUSc). The minimum inference time required to give good behavior is generally 8-10Hz.

**Comment by mandelyoni** [comment]
Hi Antonio,
In your paper, DroNet: Learning to Fly by Driving, in Table 3, you make a comparison with Gandhi et al.
Did you also implement their algorithm?
Thank you,  Yoni

**Comment by antonilo** [comment]
Yes, we've re-implemented their algorithm as described in the paper.

**Comment by mandelyoni** [comment]
Ok, thank you.

**Comment by mandelyoni** [comment]
Hi Antonio, 
Did you run Ubuntu 16.04 with ROS on the Upboard?
Is the simple version of the board adequate (https://up-shop.org/up-boards/272-up-board.html)?
Thanks, Yoni
